controllers/subjectpermission/subjectpermission_controller.go

```diff
@@ -1,78 +1,63 @@
-	"sigs.k8s.io/controller-runtime/pkg/handler"
 	logf "sigs.k8s.io/controller-runtime/pkg/log"
-	"sigs.k8s.io/controller-runtime/pkg/manager"
-	"sigs.k8s.io/controller-runtime/pkg/reconcile"
-	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	managedv1alpha1 "github.com/openshift/rbac-permissions-operator/api/v1alpha1"
 )
 
 var log = logf.Log.WithName("controller_subjectpermission")
 
-/**
-* USER ACTION REQUIRED: This is a scaffold file intended for the user to modify with their own Controller
-* business logic.  Delete these comments after modifying this file.*
- */
-
-// Add creates a new SubjectPermission Controller and adds it to the Manager. The Manager will set fields on the Controller
-// and Start it when the Manager is Started.
-func Add(mgr manager.Manager) error {
-	return add(mgr, newReconciler(mgr))
-}
-
-// newReconciler returns a new reconcile.Reconciler
-func newReconciler(mgr manager.Manager) reconcile.Reconciler {
-	return &ReconcileSubjectPermission{Client: mgr.GetClient(), Scheme: mgr.GetScheme()}
-}
-
-// add adds a new Controller to mgr with r as the reconcile.Reconciler
-func add(mgr manager.Manager, r reconcile.Reconciler) error {
-	// Create a new controller
-	c, err := controller.New("subjectpermission-controller", mgr, controller.Options{Reconciler: r})
-	if err != nil {
-		return err
-	}
-
-	// Watch for changes to primary resource SubjectPermission
-	err = c.Watch(&source.Kind{Type: &managedv1alpha1.SubjectPermission{}}, &handler.EnqueueRequestForObject{})
-	if err != nil {
-		return err
-	}
-
-	return nil
-}
-
-// blank assignment to verify that ReconcileSubjectPermission implements reconcile.Reconciler
-var _ reconcile.Reconciler = &ReconcileSubjectPermission{}
-
-// ReconcileSubjectPermission reconciles a SubjectPermission object
-type ReconcileSubjectPermission struct {
-	// This client, initialized using mgr.Client() above, is a split client
-	// that reads objects from the cache and writes to the apiserver
-	Client client.Client
+// SubjectPermissionReconciler reconciles a SubjectPermission object
+type SubjectPermissionReconciler struct {
+	client.Client
 	Scheme *runtime.Scheme
 }
 
-// Reconcile reads that state of the cluster for a SubjectPermission object and makes changes based on the state read
-// and what is in the SubjectPermission.Spec
-// Note:
-// The Controller will requeue the Request to be processed again if the returned error is non-nil or
-// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
-func (r *ReconcileSubjectPermission) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+//+kubebuilder:rbac:groups=managed.openshift.io,resources=subjectpermissions,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=managed.openshift.io,resources=subjectpermissions/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=managed.openshift.io,resources=subjectpermissions/finalizers,verbs=update
```

Comment:
Same query as for namespace controller rbac markers too. Could you confirm we need these?